misk/src/main/kotlin/misk/MiskServiceModule.kt

```diff
@@ -131,17 +107,39 @@ class MiskCommonServiceModule : KAbstractModule() {
     }
 
     val builder = ServiceGraphBuilder()
+
+    // Support the deprecated DependantService interface.
+    for (service in services) {
+      var key : Key<*>
+      when (service) {
+        is DependentService -> {
+          key = service.producedKeys.firstOrNull() ?: service::class.toKey()
+          for (consumedKey in service.consumedKeys) {
+            builder.addDependency(dependent = key, dependsOn = consumedKey)
+          }
+        }
+        else -> {
+          key = service::class.toKey()
```

Comment:
Can you change addService() to explode if the same key is added multiple times? I’m worried about name collisions here but that should protect against it.

Reply:
Done!